cache/compressed_secondary_cache.cc

```diff
@@ -271,7 +293,12 @@ Status CompressedSecondaryCache::Insert(const Slice& key,
 Status CompressedSecondaryCache::InsertSaved(
     const Slice& key, const Slice& saved, CompressionType type = kNoCompression,
     CacheTier source = CacheTier::kVolatileTier) {
-  if (type == kNoCompression) {
+  if (type == kNoCompression || source == CacheTier::kVolatileCompressedTier) {
+    assert(source != CacheTier::kVolatileCompressedTier);
```

Comment:
This change is a bit hard to understand. Why not have this assert outside the if block? Source should never be kVolatileCompressedTier?